### ISSUE TITLE
Harvester Integration - single Harvester cluster UI

### ISSUE THREAD

**Issue by yasker** [rancher/dashboard#3894]
We need to merge the Harvester UI into the Rancher dashboard.

**Comment by gaktive** [comment]
Once https://github.com/rancher/dashboard/pull/3726 is merged, then UI can work on this.

**Comment by gaktive** [comment]
https://github.com/rancher/dashboard/pull/3881 is also needed here; @vincent99 has provided some review but questions remain.

**Comment by WuJun2016** [comment]
@guangbochen  @yasker  It is a good suggestion to distinguish between rancher and harvester in the code. At present, we use the virtual keyword to distinguish. In @richard-cox's comment, I saw that he suggested using harvester or hci as the keyword distinction will be more specific。   Do you have any suggestions for this?

This more specific name will change the url below,
https://172.16.1.245:30443/dashboard/c/local/virtual/volume, 
the virtual in this url will become the specific name we specify


cc @richard-cox

**Comment by yasker** [comment]
I am fine with `harvester`

**Comment by slickwarren** [comment]
There are a few pages not working on v2.6-head (053d4c4ea4ce8218a55fc7efc34f22209239211c) for **standard users** :
* **Reopening for the following: What permissions should a standard user have in virtualization management?**
* some pages fail with the following error:
<img width="1362" alt="Screen Shot 2021-09-22 at 11 00 40 AM" src="https://user-images.githubusercontent.com/16691014/134397346-9512eac8-9d37-493d-a494-9c0308a408aa.png">

* while others will still show to the user, but are not functional and either break the UI after trying to use the page, or don't display properly to begin with. i.e.
  * after adding a cluster member, the UI appears to break. even a hard refresh + reload did not resolve this on `<hostname>/dashboard/c/c-m-6kr82bzn/harvester/members` page:
  * <img width="1790" alt="Screen Shot 2021-09-22 at 11 04 07 AM" src="https://user-images.githubusercontent.com/16691014/134397605-ffea475a-b85b-4da9-b8fc-6e52f3630d93.png">

* no resources are displaying for standard users.

**Comment by yasker** [comment]
cc @WuJun2016 @noahgildersleeve @guangbochen @gitlawr 

@slickwarren User should be able to create VM/images in their own project/namespace. I am not sure if cluster member should be shown to normal users without privilege acess

**Comment by richard-cox** [comment]
The dashboard's `harvester` product needs RBAC applied to it's sub-`products` and `virtualTypes` via the `ifHaveType` property. It's there for some, not for others. This would then remove them from the side menu.

**Comment by gitlawr** [comment]
1. IIRC, we should show namespaced resource page like VM/Volume/Image and hide cluster-scoped resources like Hosts/Namespaces when users don't have permissions
2. I get the type-not-found error as a cluster owner(standard user). Permissions/Harvester APIs are available. It works for the admin user.
3. I can reproduce the cluster members page issue in native explorer: https://github.com/rancher/dashboard/issues/4263

**Comment by gaktive** [comment]
@yasker I see two items were merged here. Is there any additional work left to do for either Harvester or UI?

**Comment by yasker** [comment]
@slickwarren Are you still being blocked or you can validate the fixes for this issue?

**Comment by slickwarren** [comment]
tested as a standard user on v2.6-head (06e19b0):
* projects/namespaces are not listing in the UI `TypeError: Cannot read properties of undefined (reading 'namespace')`
<img width="1771" alt="Screen Shot 2021-09-28 at 9 47 06 AM" src="https://user-images.githubusercontent.com/16691014/135130579-d03e56b1-24fc-4e78-a013-6e94d1f20323.png">

* RBAC page is not populating users (expecting it to at least show current user viewing the project)

<img width="1767" alt="Screen Shot 2021-09-28 at 9 49 45 AM" src="https://user-images.githubusercontent.com/16691014/135131153-e58ca08f-ca00-4c32-8f9f-7aba8904c9ae.png">

* `create` page for `cloud config templates` shows the raw type, `config map` . This title should match the type, `cloud config template`

<img width="1363" alt="Screen Shot 2021-09-28 at 9 50 55 AM" src="https://user-images.githubusercontent.com/16691014/135131307-1db46cc7-5081-47cf-a8e6-9a46764f1f0e.png">

* `hosts` page is also not displaying any information for the user

<img width="1118" alt="Screen Shot 2021-09-28 at 9 54 01 AM" src="https://user-images.githubusercontent.com/16691014/135131721-26ba76eb-2178-45c2-9ab1-b9350bae92c2.png">

**Comment by slickwarren** [comment]
closing in favor of https://github.com/rancher/dashboard/issues/4299
